src/Cloud/DecoratorPluginManager.php

```diff
@@ -40,4 +41,37 @@ class DecoratorPluginManager extends AbstractPluginManager
     ];
 
     protected $instanceOf = Decorator\DecoratorInterface::class;
+
+    /**
+     * Validate the plugin is of the expected type (v3).
+     *
+     * Validates against `$instanceOf`.
+     *
+     * @param mixed $instance
+     * @throws InvalidServiceException
+     */
+    public function validate($instance)
+    {
+        if (! $instance instanceof $this->instanceOf) {
+            throw new InvalidServiceException(sprintf(
+                '%s can only create instances of %s; %s is invalid',
+                get_class($this),
+                $this->instanceOf,
+                (is_object($instance) ? get_class($instance) : gettype($instance))
+            ));
+        }
+    }
+
+    /**
+     * Validate the plugin is of the expected type (v2).
+     *
+     * Proxies to `validate()`.
+     *
+     * @param mixed $instance
+     * @throws InvalidServiceException
+     */
+    public function validatePlugin($instance)
+    {
+        $this->validate($instance);
```

Comment:
Take a look at zendframework/zend-servicemanager#93 — the author details having `validatePlugin()` do a try/catch around the `validate()` call, to allow it to throw the component-specific exception for v2.

We don't 100% need to do that, but it would ensure that *if* users are doing try/catch around calls to the plugin manager, those don't break.